apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -379,39 +378,23 @@ defmodule Aecore.Chain.Worker do
   defp calculate_block_acc_txs_info(block) do
     block_hash = BlockValidation.block_header_hash(block.header)
 
-    accounts =
-      for tx <- block.txs do
-        case tx.data do
-          %SpendTx{} ->
-            [tx.data.sender, tx.data.receiver]
-
-          %DataTx{} ->
-            tx.data.sender
-        end
-      end
-
-    accounts_unique = accounts |> List.flatten() |> Enum.uniq() |> List.delete(nil)
+    accounts_unique =
+      block.txs
+      |> Enum.map(&[&1.data.sender, &1.data.payload.receiver])
```

Comment:
by removing the pattern match, we make it more difficult to use this with transactions that have no receiver.